### PR TITLE
implement occupation cache

### PR DESCRIPTION
**Pull request by chilliwilly77**

### CODE REVIEW

File: app/controllers/occupation-controller.js [hunk 1]
```diff
@@ -1,35 +1,25 @@
...
 
-/* GET home page. */
-function capitalizeFirstLetter(string) {
-  return string.charAt(0).toUpperCase() + string.slice(1);
-}
+const capitalizeFirstLetter = string =>
+  string.charAt(0).toUpperCase() + string.slice(1);
+
+const occupationViewModel = occupation => Object.assign(occupation, {
+  tasks: occupation.tasks.split('; ').map(s => capitalizeFirstLetter(s)),
```
Comment (will518) [inline app/controllers/occupation-controller.js]: Can we remove the `s =>` if the function passed to map takes one parameter?

File: app/services/lmi-client-service.js [hunk 2]
```diff
@@ -1,49 +1,85 @@
 const config = require('./../config/index');
 const request = require('request-promise-native');
 const crypto = require('crypto');
+const OccupationCache = require('./../models/occupation-cache-model');
+
+const filterObjectByKeys = function (body, keys) {
+  return Object.keys(body)
+    .filter(key => keys.indexOf(key) > -1)
```
Comment (will518) [inline app/services/lmi-client-service.js]: Can we use `keys.includes()` here?

File: app/services/lmi-client-service.js [hunk 3]
```diff
@@ -1,49 +1,85 @@
...
     return this.lmiGetRequest(`/ashe/estimatePay?soc=${socCode}`)
       .then(response => this.handleAsheResponse(response, socCode, 'estpay'));
   }
 
+  toOccupationCacheDTO(socCode, soc, weekHours, weekPay) {
+    const body = soc.body && JSON.parse(soc.body);
+    if (!body) {
+      throw new Error(`${socCode} not found`);
+    }
+    body.additionalTitles = body.add_titles.join('; ');
```
Comment (will518) [inline app/services/lmi-client-service.js]: Should we pull out the '; ' separator as it is used in the occupation-controller.js as well?

Comment (chilliwilly77) [reply app/services/lmi-client-service.js]: confusingly it is two separate fields that use a semi colon separator. additionalTitles and tasks. This is for the DTO translation of additionalTitles property from API to DB, and the one in the occupation controller is a view transformation of the tasks property... so I think it is ok in this instance

**Comment by will518** [comment]
👍

**Pull request merged**